### PR TITLE
Webgl line

### LINKED ISSUE #2681: Webgl line glyph
The line glyph is one of the glyphs that should definitely have webgl support to allow rendering large datasets. This is a bit more involved than marker glyphs, because the line needs to be drawn using triangles. Standard webgl lines do not officially support widths other than 1, and have poor anti-aliasing. Let alone stippling and caps/joins.

I'm implementing this technique by Nicolas Rougier: http://jcgt.org/published/0002/02/08/ for which there is code in glumpy and vispy.

Line features we need to support (Bokeh basically supports whatever 2D canvas supports):

* color and alpha - ok, can use css-to-rgb conversion that I added earlier
* width - ok
* join/cap - 'butt', 'round', 'square' - ok, the implemenation supports these and more
* dash - ok, spec seems the same
* dash_offset - ok

(conclusion: it seems we can provide full support of lines with webgl)

Tasks/components:

* [x] bake vertices based on input data (i.e. convert line to triangles)
* [x] unfolding (to prevent artefacts at sharp corners)
* [x] create dash atlas
* [x] vertex and fragment shader

### PR DESCRIPTION
**Pull request by almarklein**
issues: closes #2681

Implements a WebGL version of the line glyph. Everything works, but there are some glitches that appear under some circumstances. They are probably there because I did not manage to make it deal with anisotropic scaling exactly right. Spend quite some time on that now. Should revisit later. The glitches appear when using dashing and using very long dash periods or very thick (16+) lines.

This also adds two new examples: one line example to show 10k points, and one to compare the canvas and webgl-drawn line side-by-side.

This is in principle done and ready for review. Minor issues to do:
* [x] Test on IE (will do tomorrow)
* [x] Edit docs.

cc @canavandl

### CODE REVIEW

**Comment by almarklein** [comment]
screenshot from 2015-10-15 00 09 32

File: bokehjs/src/coffee/renderer/glyph/bokehgl.coffee [hunk 1]
```diff
@@ -84,6 +85,72 @@ attach_color = (prog, vbo, att_name, n, visual) ->
...
+    @tex.set_size([@_height, @_width], gl.RGBA)
+    # Init with solid line (index 0 is reserved for this)
+    @get_atlas_data([1])
+  
+  get_atlas_data: (pattern) ->
+    key = pattern.join('-')
+    findex_period = @_atlas[key]
+    if findex_period is undefined
+      [data, period] = @make_pattern(pattern)
+      #console.log(period, key, data)
```
Comment (canavandl) [inline bokehjs/src/coffee/renderer/glyph/bokehgl.coffee]: Missed console.log statement

**Comment by canavandl** [comment]
The webgl line implementation appears to work. All of the bokeh LineProps are supported and I was able to render >100k points.

Needs an example in the user_guide/webgl docs. Other than that, LGTM

**Comment by almarklein** [comment]
Addressed comments. cc @bryevdv

**Comment by bryevdv** [comment]
This still is not working on Safari, but that's a separate issue. FYI there is a different message on newer safari: 

screen shot 2015-10-31 at 7 47 12 pm

Apart from that works great. I think at some point we might want to discuss whether the GL-specific code can be optionally included. ping @mattpap

**Pull request merged**